gammapy/image/maps.py

```diff
@@ -416,13 +417,18 @@ def lookup_max(self, region=None):
 
     def solid_angle(self):
         """
-        Solid angle image
+        Solid angle image (2-dim `astropy.units.Quantity` in `sr`).
         """
         coordinates = self.coordinates(mode='edges')
-        xsky = coordinates.data.lon
-        ysky = coordinates.data.lat
-        omega = np.abs(np.diff(xsky, axis=1)[1:, :] * np.diff(ysky, axis=0)[:, 1:])
-        return omega.to('sr')
+        lon = coordinates.data.lon.radian
+        lat = coordinates.data.lat.radian
+
+        dx = angular_separation(lon[:, :-1], lat[:, :-1],
+                                    lon[:, 1:], lat[:, :-1])
+
+        dy = angular_separation(lon[:-1, :], lat[:-1, :],
+                                lon[1:, :], lat[1:, :])
+        return Quantity(dx[1:, :] * dy[:, 1:], "sr")
```

Comment:
@adonath - The current implementation does the slicing along the two axes in two steps. It's probably correct, but I agree it's hard to read and should be changed.
How about this implementation https://github.com/gammapy/gammapy/pull/568#issuecomment-227069333 ?